jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/UserControllerTest.groovy

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (C) 2011  JTalks.org Team
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.jtalks.jcommune.test
+
+import org.jtalks.common.model.entity.Component
+import org.jtalks.common.model.permissions.GeneralPermission
+import org.jtalks.jcommune.service.security.AdministrationGroup
+import org.jtalks.jcommune.test.model.User
+import org.jtalks.jcommune.test.service.ComponentService
+import org.jtalks.jcommune.test.service.GroupsService
+import org.jtalks.jcommune.test.utils.Users
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.MediaType
+import org.springframework.mock.web.MockHttpSession
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.transaction.TransactionConfiguration
+import org.springframework.test.context.web.WebAppConfiguration
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.transaction.annotation.Transactional
+import spock.lang.Specification
+
+import static org.hamcrest.Matchers.hasItems
+import static org.hamcrest.Matchers.is
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
+
+/**
+ * @author skythet
+ */
+@WebAppConfiguration
+@ContextConfiguration(locations = 'classpath:/org/jtalks/jcommune/web/view/test-configuration.xml')
+@TransactionConfiguration(transactionManager = "transactionManager", defaultRollback = true)
+@Transactional
```

Comment:
If drop this, , it will be a error:
`org.springframework.orm.hibernate3.HibernateSystemException: No Hibernate Session bound to thread, and configuration does not allow creation of non-transactional one here; nested exception is org.hibernate.HibernateException: No Hibernate Session bound to thread, and configuration does not allow creation of non-transactional one here`